tensorflow_addons/optimizers/lamb.py

```diff
@@ -0,0 +1,248 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""LAMB (Layer-wise Adaptive Moments) optimizer as TF2 tf.keras.optimizers.
+
+See paper [Large Batch Optimization for Deep Learning: Training BERT in
+76 minutes](https://arxiv.org/abs/1904.00962).
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import re
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+class LAMB(tf.keras.optimizers.Optimizer):
+    """Optimizer that implements the LAMB (Layer-wise Adaptive Moments)
+    optimizer as TF2 tf.keras.optimizers.
+
+    See paper [Large Batch Optimization for Deep Learning: Training BERT
+    in 76 minutes](https://arxiv.org/abs/1904.00962).
+    """
+
+    def __init__(self,
+                 learning_rate=0.001,
+                 beta_1=0.9,
+                 beta_2=0.999,
+                 epsilon=1e-6,
+                 weight_decay_rate=0.0,
+                 exclude_from_weight_decay=None,
+                 exclude_from_layer_adaptation=None,
+                 name='LAMB',
+                 **kwargs):
+        """
+        learning_rate: A `Tensor` or a floating point value.
+          The learning rate.
+        beta_1: A `float` value or a constant `float` tensor.
+          The exponential decay rate for the 1st moment estimates.
+        beta_2: A `float` value or a constant `float` tensor.
+          The exponential decay rate for the 2nd moment estimates.
+        epsilon: A small constant for numerical stability.
+        weight_decay_rate: weight decay rate.
+        exclude_from_weight_decay: comma separated name patterns of variables
+          excluded from weight decay. Variables whose name contain a substring
+          matching the pattern will be excluded.
+        exclude_from_layer_adaptation: comma separated name patterns of
+          variables excluded from layer adaptation. Variables whose name contain
```

Comment:
this line to long (len == 80). Please lint it manually as autopep8 could not format it.